File: fingertip/src/main/java/com/praszapps/fingertip/model/di/module/FingertipRepoModule.java
package com.praszapps.fingertip.model.di.module;

import com.praszapps.fingertip.model.repository.FingertipRepository;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class FingertipRepoModule {

    @Provides
    @Singleton
    public FingertipRepository provideRepository() {
        return new FingertipRepository();
    }
}

File: fingertip/src/main/java/com/praszapps/fingertip/model/di/module/FingertipRepoModule.kt
package com.praszapps.fingertip.model.di.module

import com.praszapps.fingertip.model.repository.FingertipRepository
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class FingertipRepoModule {

    @Provides
    @Singleton
    fun provideRepository(): FingertipRepository {
        return FingertipRepository()
    }
}